### DISCUSSION TITLE
decimal number bug

### DISCUSSION THREAD

**Discussion by Etkiny** [Budibase/budibase#5482] [Help]
Hello guys,

I think there is a bug about numbers. I have a query which updates decimal(10,2) number on my table in DB. The problem is when I try to update the number as 10,90 the result comes as 2001, 2010, or 70,90 comes out as 1970.00, 1990.00 etc.  The result comes true once or twice of my several tryings. I thought it might be a bug.

**Comment by melohagan** [comment]
Hello @Etkiny 

Would you be able to provide the query code snippet? 

It seems to me like something is being treated as a date type rather than a number.

**Reply by Etkiny** [reply]
Sure. @melohagan 

`Update Product p left join ProductStock ps on p.productNumber = ps.productNumber
    SET p.price = {{productPrice}},
        ps.fullPrice = {{productPrice}},
        ps.price = {{cutPrice}},
        ps.maximumQuantity = {{maxQuantity}}

where bp.productNumber= {{productNumber}};` 

Their types are number on schema. The result comes true sometimes, though.  Also, when I enter the value as "19,90" or "10.90" the value changes into "10,9" .and this happens only  with the numbers ending wiht "90" like "x,90".  I am okay with 10,80 or 40,10 etc.

**Reply by melohagan** [reply]
Is it saving those values to the SQL database incorrectly, or is it that the Budibase app is reading the values incorrectly? 
Are the binding values coming from text form fields?
Also, which RDBMS are you using?

**Reply by isakulaksiz** [reply]
@Etkiny you should use ORM instead of writing sql query directly.
Like Hibernate or Entity Framework Core

**Reply by Etkiny** [reply]
> Is it saving those values to the SQL database incorrectly, or is it that the Budibase app is reading the values incorrectly? Are the binding values coming from text form fields? Also, which RDBMS are you using?

Yes, It saves the values incorrectly Binding values are not coming from text fields. I am using MySQL. It is not always like that, sometimes it happens and under what conditions happening is hard to detect. It seems like it is random

**Reply by melohagan** [reply]
Hey @Etkiny, Would you be able to share the request body that's sent whenever you click your 'Save' button. 
Also it would be good to know what type of form elements you are using, as well as a screenshot of how the query is being used in your design section.
